apps/src/block_utils.js

```diff
@@ -577,7 +594,14 @@ const determineInputs = function (text, args, strictTypes = []) {
     const label = parts[1];
     const inputName = parts[3];
     if (inputName) {
-      const arg = findAndRemoveInputConfig(args, inputName);
+      const arg = findAndRemoveInputConfig(args, inputName, text);
+      if (arg === null) {
+        // If no valid arg was found, just use the label.
+        return {
+          mode: DUMMY_INPUT,
+          label,
+        };
+      }
```

Comment:
I was wondering about the case when the number of args > input variables. If we keep count of the number of input variables and compare with args length, we could log that there was a mismatch similar to what you do in line 566. This seems unlikely but seems like the counter case of one above.

Reply:
So I believe we already catch and log that case here: https://github.com/code-dot-org/code-dot-org/blob/a113fa76d7e2a761c2fdc901bff83167e28aa5ef/apps/src/block_utils.js#L632 I could turn this into a MetricsReporter call if we think that's useful to track?

Reply:
Oh cool. I think either the current warning or `MetricsReporter` call works! But maybe for the sake of consistency, we could use`MetricsReporter` since you're adding it for the counter case.

Reply:
Updated to use MetricsReporter!

Reply:
Amazing coincidence that this bug [was reported](https://codedotorg.slack.com/archives/C05P12U42F5/p1702651531649469) today! I think using `MetricsReporter` is the right call - thank you!